nixos/modules/virtualisation/railcar.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.railcar;
+in
+{
+  options.services.railcar = {
+    enable = mkEnableOption "railcar";
+
+    containers = mkOption {
+      default = {};
+      type = with types; loaOf (submodule ({ name, config, ... }: {
+        options = {
+          cmd = mkOption {
+            type = types.package;
+            description = "Command to run inside the container";
+          };
+
+          packages = mkOption {
+            type = with types; loaOf (package);
+            default = [];
+            description = "A set of additional packages to provide";
+          };
```

Comment:
What's the use case for this, over referencing these packages inside your main derivation?

Reply:
I'm mostly thinking this could be useful for debugging purposes

Reply:
What sort of debugging purposes? AFAICT, there's nowhere you couldn't just use a buildEnv to throw in extra packages.

Reply:
Yea, realistically the idea was to have that option be a prettier way of adding packages to a buildEnv. Also to separate "additional packages in the container" and "the command that's run".

Reply:
I'm still not convinced this is a good idea. I think having two different ways of adding packages in the container will make things difficult to debug, not easier, and doing this means that you're forced to have a `buildEnv` even if you only need a single package.

I think the reason I'm so uncomfortable with this is that it feels like a way to globally install packages, which we shouldn't ever need to do in an immutable environment entirely built with Nix.